src/docker-osx-dev

```diff
@@ -829,10 +829,17 @@ function initial_sync {
   log_info "Performing initial sync of paths: ${paths_to_sync[@]}"
 
   local dirs_to_create=()
+  local parent_dirs=()
+  local base_names=()
   local path=""
+  local parent_dir=""
```

Comment:
I'm not entirely sure about scoping in bash, but since they're used in 2 loops in a single code block, wouldn't that fail? I don't think do...done has its own scope.

also, it's not really read-only, is it? since every iteration of the loop changes them?